compiler/nim.nim

```diff
@@ -7,21 +7,26 @@
 #    distribution, for details about the copyright.
 #
 
-when defined(gcc) and defined(windows):
-  when defined(x86):
-    {.link: "../icons/nim.res".}
-  else:
-    {.link: "../icons/nim_icon.o".}
+import std/[os, strutils, parseopt]
+when defined(windows):
+  const dir = when defined(nimKochBootstrap):
+    # remove workaround pending bootstrap >= 1.5.1
+    # refs https://github.com/nim-lang/Nim/issues/18334#issuecomment-867114536
+    currentSourcePath.parentDir.quoteShell & "\\"
```

Comment:
Quoting only a part of the path is bad style. Why not simply:

```nim

when defined(amd64) and defined(windows) and defined(vcc) and not defined(nimKochBootstrap):
  {.link: "../icons/nim-amd64-windows-vcc.res".}
when defined(i386) and defined(windows) and defined(vcc) and not defined(nimKochBootstrap):
  {.link: "../icons/nim-i386-windows-vcc.res".}

```

Reply:
done;
but i had done it that way to preserve semantics (would be needed anyways if those files were not optional); in future if there's a situation where those are not optional, smthg like this should work:
```nim
template link2(a) =
  when defined(nimKochBootstrap):
    const a2 = quoteShell(currentSourcePath.parentDir / a)
    {.link: a2.}
else:
  {.link: a.}
```
then use link2 in such places